app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -756,10 +754,13 @@ private void launchNextActivityOnAppInstall() {
                 Intent i = new Intent(getApplicationContext(), DispatchActivity.class);
                 startActivity(i);
             }
-        } else {
+        } else if (getCallingActivity().getPackageName().startsWith(BuildConfig.APPLICATION_ID)){

```

Comment:
this is crashing `AppDependenciesTest` with - 

````
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.ComponentName.getPackageName()' on a null object reference
	at org.commcare.activities.CommCareSetupActivity.launchNextActivityOnAppInstall(SourceFile:757)
	at org.commcare.activities.CommCareSetupActivity.onResume(SourceFile:297)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1361)
````

We will need to do a null check on `getCallingActivity()`

Reply:
Also is there a reason to not exact match the package strings ?

Reply:
Not a very strong reason, just wanted to be able to respond to calls from other apps in the same domain. Should we make this stricter?

Reply:
Yeah I think we should do a strict match as we should only come here using internal launch from CommCare itself. All other calls would be external and should use VIEW action on intent.